libs/langchain-community/src/embeddings/googlegenerativeai.ts

```diff
@@ -0,0 +1,215 @@
+   *
+   * Note: The format must follow the pattern - `{model}`
+   */
+  modelName?: string;
+
+  /**
+   * Type of task for which the embedding will be used
+   *
+   * Note: currently only supported by `embedding-001` model
+   */
+  taskType?: TaskType;
+
+  /**
+   * An optional title for the text. Only applicable when TaskType is
+   * `RETRIEVAL_DOCUMENT`
+   *
+   * Note: currently only supported by `embedding-001` model
+   */
+  title?: string;
+
+  /**
+   * Whether to strip new lines from the input text. Default to true
+   */
+  stripNewLines?: boolean;
+
+  /**
+   * Google API key to use
+   */
+  apiKey?: string;
+}
+
+/**
+ * Class that extends the Embeddings class and provides methods for
+ * generating embeddings using the Google Palm API.
+ * @example
+ * ```typescript
+ * const model = new GoogleGenerativeAIEmbeddings({
+ *   apiKey: "<YOUR API KEY>",
+ *   modelName: "embedding-001",
+ * });
+ *
+ * // Embed a single query
+ * const res = await model.embedQuery(
+ *   "What would be a good company name for a company that makes colorful socks?"
+ * );
+ * console.log({ res });
+ *
+ * // Embed multiple documents
+ * const documentRes = await model.embedDocuments(["Hello world", "Bye bye"]);
+ * console.log({ documentRes });
+ * ```
+ */
+export class GoogleGenerativeAIEmbeddings
+  extends Embeddings
+  implements GoogleGenerativeAIEmbeddingsParams
+{
+  apiKey?: string;
+
+  modelName = "embedding-001";
+
+  taskType?: TaskType;
+  title?: string;
+  stripNewLines: boolean = true;
+
+  private client: GenerativeModel | TextServiceClient;
+
+  get _isEmbedContentModel() {
+    if (/^embedding-\d+$/.test(this.modelName)) {
+      return true;
+    }
+    return false;
+  }
+
+  constructor(fields?: GoogleGenerativeAIEmbeddingsParams) {
+    super(fields ?? {});
+
+    this.modelName =
+      fields?.modelName?.replace(/^models\//, "") ?? this.modelName;
+
+    const isEmbedContentModel = this._isEmbedContentModel;
+
+    this.taskType = fields?.taskType ?? this.taskType;
+    if (this.taskType && !isEmbedContentModel) {
+      throw new Error(
+        "Task type is only supported for models with `embedContent` method"
+      );
+    }
+
+    this.title = fields?.title ?? this.title;
+    if (this.title && !isEmbedContentModel) {
+      throw new Error(
+        "Title is only supported for models with `embedContent` method"
+      );
+    }
+    if (this.title && this.taskType !== TaskType.RETRIEVAL_DOCUMENT) {
+      throw new Error(
+        "title can only be sepcified with TaskType.RETRIEVAL_DOCUMENT"
+      );
+    }
+
+    this.apiKey = fields?.apiKey ?? getEnvironmentVariable("GOOGLE_API_KEY");
+    if (!this.apiKey) {
+      throw new Error(
+        "Please set an API key for Google GenerativeAI " +
+          "in the environmentb variable GOOGLE_API_KEY " +
+          "or in the `apiKey` field of the " +
+          "GoogleGenerativeAIEmbeddings constructor"
+      );
+    }
+
+    this.client = isEmbedContentModel
```

Comment:
Is the difference here multimodal vs. text embeddings?

Reply:
There are different methods with different arguments for different models:
for `embedding-001` it's  `embedContent` and for `embedding-gecko-002` it's `embedText`